src/app/clusters/window-covering-server/window-covering-server.cpp

```diff
@@ -158,26 +167,42 @@ bool IsOpen(chip::EndpointId endpoint)
     uint16_t liftLimit    = 0;
     uint16_t tiltPosition = 0;
     uint16_t tiltLimit    = 0;
+    bool isOpen           = false;
+
+    if (HasFeature(endpoint, WindowCoveringFeature::kLift) && HasFeature(endpoint, WindowCoveringFeature::kPositionAwareLift) &&
+        HasFeature(endpoint, WindowCoveringFeature::kAbsolutePosition) &&
+        EMBER_ZCL_STATUS_SUCCESS == Attributes::TargetPositionLiftPercent100ths::Get(endpoint, &liftPosition) &&
+        EMBER_ZCL_STATUS_SUCCESS == Attributes::InstalledOpenLimitLift::Get(endpoint, &liftLimit))
+    {
+        isOpen = liftPosition == liftLimit;
+    }
+    else if (HasFeature(endpoint, WindowCoveringFeature::kLift) &&
+             HasFeature(endpoint, WindowCoveringFeature::kPositionAwareLift) &&
+             EMBER_ZCL_STATUS_SUCCESS == Attributes::TargetPositionLiftPercent100ths::Get(endpoint, &liftPosition))
+    {
+        isOpen = 0 == liftPosition;
+    }
+
+    if (HasFeature(endpoint, WindowCoveringFeature::kTilt) && HasFeature(endpoint, WindowCoveringFeature::kPositionAwareTilt) &&
+        HasFeature(endpoint, WindowCoveringFeature::kAbsolutePosition) &&
+        EMBER_ZCL_STATUS_SUCCESS == Attributes::TargetPositionTiltPercent100ths::Get(endpoint, &tiltPosition) &&
+        EMBER_ZCL_STATUS_SUCCESS == Attributes::InstalledOpenLimitTilt::Get(endpoint, &tiltLimit))
+    {
+        isOpen = isOpen && tiltPosition == tiltLimit;
+    }
+    else if (HasFeature(endpoint, WindowCoveringFeature::kTilt) &&
+             HasFeature(endpoint, WindowCoveringFeature::kPositionAwareTilt) &&
+             EMBER_ZCL_STATUS_SUCCESS == Attributes::TargetPositionTiltPercent100ths::Get(endpoint, &tiltPosition))
+    {
+        isOpen = isOpen && 0 == tiltPosition;
+    }
 
-    Attributes::TargetPositionLiftPercent100ths::Get(endpoint, &liftPosition);
-    Attributes::InstalledOpenLimitLift::Get(endpoint, &liftLimit);
-    Attributes::TargetPositionTiltPercent100ths::Get(endpoint, &tiltPosition);
-    Attributes::InstalledOpenLimitTilt::Get(endpoint, &tiltLimit);
-    return liftPosition == liftLimit && tiltPosition == tiltLimit;
+    return isOpen;
 }
 
 bool IsClosed(chip::EndpointId endpoint)
 {
-    uint16_t liftPosition = 0;
```

Comment:
I wanted to get rid of those IsClosed IsOpen 
There are wrong in the previous implementation and in this one 
You don't need to check the absolute limits

Reply:
I was working on a cleaner way to handle this , it might be preferable to drop those IMO, or we merge and then i ll remove them later

Reply:
I will just stick to it. It can be removed afterwords.